### ISSUE TITLE
500 Server Error after clean install in VPS

### ISSUE THREAD

**Issue by Afan417** [vitodeploy/vito#38]
I followed the installation instructions of the docs, I installed it in a Digital Ocean 512mb and 1vCPU with 10GB Disk space, and just got a default 500 Server Error after running command specified in docs:
 (``wget https://raw.githubusercontent.com/vitodeploy/vito/main/install/install.sh && bash install.sh``)

I tried to test if I can login to mysql but it gave me this error, can't seem to figure out why it's not working, I got all the green OK checkmarks after running the initial install command, what else do I need to do?

ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)

Here's my .env file, didn't change anything, default values:
```
APP_NAME=Vito
APP_ENV=production
APP_KEY=base64:kXgYXYDEzpSMjZVTUmB2QfnC1R/YugrzHS249zB78Ag=
APP_DEBUG=false
APP_URL=https://vito.afanajdari.tech

LOG_CHANNEL=stack
LOG_LEVEL=debug

DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=vito
DB_USERNAME=vito
DB_PASSWORD=7dDVf5Qh84EKs9sv

BROADCAST_DRIVER=pusher
CACHE_DRIVER=file
FILESYSTEM_DRIVER=local
QUEUE_CONNECTION=default
SESSION_DRIVER=database
SESSION_LIFETIME=120

REDIS_HOST=127.0.0.1
REDIS_PASSWORD=null
REDIS_PORT=6379

MAIL_MAILER=smtp
MAIL_HOST=mailhog
MAIL_PORT=1025
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS=null
MAIL_FROM_NAME="${APP_NAME}"

AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1
AWS_BUCKET=
AWS_USE_PATH_STYLE_ENDPOINT=false

PUSHER_APP_ID=app-id
PUSHER_APP_KEY=app-key
PUSHER_APP_SECRET=app-secret
PUSHER_HOST=soketi
PUSHER_PORT=6001
PUSHER_SCHEME=http
PUSHER_APP_CLUSTER=mt1
VITE_PUSHER_APP_KEY="${PUSHER_APP_KEY}"
VITE_PUSHER_HOST="${PUSHER_HOST}"
VITE_PUSHER_PORT="${PUSHER_PORT}"
VITE_PUSHER_SCHEME="${PUSHER_SCHEME}"
VITE_PUSHER_APP_CLUSTER="${PUSHER_APP_CLUSTER}"

SSH_PUBLIC_KEY_NAME=ssh-public.key
SSH_PRIVATE_KEY_NAME=ssh-private.pem

GITHUB_CLIENT_ID=
GITHUB_CLIENT_SECRET=

ABLY_KEY=
ABLY_PUBLIC_KEY=
```

Also ran a linux command to see what services are running, only these are, don't know if anything is missing from here:

![image](https://github.com/vitodeploy/vito/assets/45122414/bdd35159-abbf-4d0e-af15-a271c3615328)

System Settings:

![image](https://github.com/vitodeploy/vito/assets/45122414/74eb09fb-5ab5-44e8-aa6c-cbdad850c9d8)

**Comment by saeedvaziry** [comment]
Can you dump your laravel.log file here?
You can find it in storage/logs

**Comment by Afan417** [comment]
Can't copy all the contents of that file, I copied the last couple lines, It keeps throwing this error non stop

```
"}
[2023-09-04 17:56:40] production.ERROR: SQLSTATE[HY000] [2002] Connection refused {"exception":"[object] (PDOException(code: 2002): SQLSTATE[HY000] [2002] Connection refused at /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php:65)
[stacktrace]
#0 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(65): PDO->__construct()
#1 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(83): Illuminate\\Database\\Connectors\\Connector->createPdoConnection()
#2 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(49): Illuminate\\Database\\Connectors\\Connector->tryAgainIfCausedByLostConnection()
#3 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/MySqlConnector.php(24): Illuminate\\Database\\Connectors\\Connector->createConnection()
#4 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(184): Illuminate\\Database\\Connectors\\MySqlConnector->connect()
#5 [internal function]: Illuminate\\Database\\Connectors\\ConnectionFactory->Illuminate\\Database\\Connectors\\{closure}()
#6 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connection.php(1210): call_user_func()
#7 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(177): Illuminate\\Database\\Connection->getPdo()
#8 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(143): Illuminate\\Database\\Connection->handleBeginTransactionException()
#9 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(117): Illuminate\\Database\\Connection->createTransaction()
#10 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(24): Illuminate\\Database\\Connection->beginTransaction()
#11 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/DatabaseQueue.php(227): Illuminate\\Database\\Connection->transaction()
#12 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(344): Illuminate\\Queue\\DatabaseQueue->pop()
#13 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(358): Illuminate\\Queue\\Worker->Illuminate\\Queue\\{closure}()
#14 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(162): Illuminate\\Queue\\Worker->getNextJob()
#15 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(137): Illuminate\\Queue\\Worker->daemon()
#16 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(120): Illuminate\\Queue\\Console\\WorkCommand->runWorker()
#17 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Queue\\Console\\WorkCommand->handle()
#18 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#19 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#20 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#21 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call()
#22 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Console/Command.php(194): Illuminate\\Container\\Container->call()
#23 /home/vito/vito.afanajdari.tech/vendor/symfony/console/Command/Command.php(312): Illuminate\\Console\\Command->execute()
#24 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Console/Command.php(164): Symfony\\Component\\Console\\Command\\Command->run()
#25 /home/vito/vito.afanajdari.tech/vendor/symfony/console/Application.php(1040): Illuminate\\Console\\Command->run()
#26 /home/vito/vito.afanajdari.tech/vendor/symfony/console/Application.php(314): Symfony\\Component\\Console\\Application->doRunCommand()
#27 /home/vito/vito.afanajdari.tech/vendor/symfony/console/Application.php(168): Symfony\\Component\\Console\\Application->doRun()
#28 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(200): Symfony\\Component\\Console\\Application->run()
#29 /home/vito/vito.afanajdari.tech/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#30 {main}
"}
[2023-09-04 17:56:46] production.ERROR: SQLSTATE[HY000] [2002] Connection refused {"exception":"[object] (PDOException(code: 2002): SQLSTATE[HY000] [2002] Connection refused at /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php:65)
[stacktrace]
#0 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(65): PDO->__construct()
#1 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(83): Illuminate\\Database\\Connectors\\Connector->createPdoConnection()
#2 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(49): Illuminate\\Database\\Connectors\\Connector->tryAgainIfCausedByLostConnection()
#3 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/MySqlConnector.php(24): Illuminate\\Database\\Connectors\\Connector->createConnection()
#4 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(184): Illuminate\\Database\\Connectors\\MySqlConnector->connect()
#5 [internal function]: Illuminate\\Database\\Connectors\\ConnectionFactory->Illuminate\\Database\\Connectors\\{closure}()
#6 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Connection.php(1210): call_user_func()
#7 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(177): Illuminate\\Database\\Connection->getPdo()
#8 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(143): Illuminate\\Database\\Connection->handleBeginTransactionException()
#9 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(117): Illuminate\\Database\\Connection->createTransaction()
#10 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(24): Illuminate\\Database\\Connection->beginTransaction()
#11 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/DatabaseQueue.php(227): Illuminate\\Database\\Connection->transaction()
#12 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(344): Illuminate\\Queue\\DatabaseQueue->pop()
#13 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(358): Illuminate\\Queue\\Worker->Illuminate\\Queue\\{closure}()
#14 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(162): Illuminate\\Queue\\Worker->getNextJob()
#15 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(137): Illuminate\\Queue\\Worker->daemon()
#16 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(120): Illuminate\\Queue\\Console\\WorkCommand->runWorker()
#17 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Queue\\Console\\WorkCommand->handle()
#18 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#19 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#20 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#21 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call()
#22 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Console/Command.php(194): Illuminate\\Container\\Container->call()
#23 /home/vito/vito.afanajdari.tech/vendor/symfony/console/Command/Command.php(312): Illuminate\\Console\\Command->execute()
#24 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Console/Command.php(164): Symfony\\Component\\Console\\Command\\Command->run()
#25 /home/vito/vito.afanajdari.tech/vendor/symfony/console/Application.php(1040): Illuminate\\Console\\Command->run()
#26 /home/vito/vito.afanajdari.tech/vendor/symfony/console/Application.php(314): Symfony\\Component\\Console\\Application->doRunCommand()
#27 /home/vito/vito.afanajdari.tech/vendor/symfony/console/Application.php(168): Symfony\\Component\\Console\\Application->doRun()
#28 /home/vito/vito.afanajdari.tech/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(200): Symfony\\Component\\Console\\Application->run()
#29 /home/vito/vito.afanajdari.tech/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#30 {main}
"}
                                                                                                                                                                           
```

**Comment by saeedvaziry** [comment]
Looks like the installation didn't finish as expected and mysql is not installed completely or damaged.

Will try to reproduce it.

But in the meanwhile you can try to reinstall and if didn't work try to allocate more memory to the server.

**Comment by Afan417** [comment]
Tomorrow night I'll try a new VPS with abit more resources and I'll let you know in this issue how it went, cheers.

**Comment by Afan417** [comment]
TLDR; It works now, I just added 512 more ram, so 1GB total, thank you for the help.

Another unrelated question, how can I add SSL, I used the first link:

![image](https://github.com/vitodeploy/vito/assets/45122414/3bcb78ab-419a-4b55-8cf4-38179525b527)

Is there any way I can add SSL, or do I need to re-install with that other command below this one?

**Comment by saeedvaziry** [comment]
Closing this issue as it doesn't look like a bug. 512mb memory is very low for having mysql 8 and other softwares together in the server.

Will update the documentation soon.
